gaphor/ui/toolbox.py

```diff
@@ -85,27 +116,55 @@ def on_collapsed(widget, prop, index):
             collapsed[index] = widget.get_property("collapsed")
             self.properties.set("toolbox-collapsed", collapsed)
 
-        for index, (title, items) in enumerate(self._toolbox_actions):
+        for index, (title, items) in enumerate(toolbox_actions):
             tool_item_group = Gtk.ToolItemGroup.new(title)
             tool_item_group.set_property("collapsed", collapsed.get(index, False))
             tool_item_group.connect("notify::collapsed", on_collapsed, index)
             for action_name, label, icon_name, shortcut, *rest in items:
-                button = toolbox_button(action_name, icon_name, label, shortcut)
+                button = self.create_toolbox_button(
+                    action_name, icon_name, label, shortcut
+                )
                 tool_item_group.insert(button, -1)
                 button.show_all()
 
             toolbox.add(tool_item_group)
             tool_item_group.show()
 
         toolbox.show()
+        return toolbox
 
-        self._toolbox = toolbox
+    def create_toolbox_container(self, toolbox: Gtk.ToolPalette) -> Gtk.ScrolledWindow:
+        """Create a toolbox container.
 
-        scrolled_window = Gtk.ScrolledWindow()
-        scrolled_window.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
-        scrolled_window.add(toolbox)
-        scrolled_window.show()
-        return scrolled_window
+        Args:
+            toolbox: The Gtk.ToolPalette to add.
+
+        Returns: The Gtk.ScrolledWindow.
+
+        """
+        toolbox_container = Gtk.ScrolledWindow()
+        toolbox_container.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.AUTOMATIC)
+        toolbox_container.add(toolbox)
+        toolbox_container.show()
+        return toolbox_container
+
+    @event_handler(ProfileSelectionChanged)
+    def _on_profile_changed(self, event: ProfileSelectionChanged) -> None:
+        """Reconfigures the toolbox based on the profile selected.
+
+        When the combo box drop down to select the profile changes
+        (UML, SysML, Safety), this event handler reconfigures the
+        toolbox.
+
+        Args:
+            event: The ProfileSelectionChanged event.
+
```

Comment:
The `Args` comment does not add a lot of value. The info is in the signature already.

Reply:
Good catch. I think a docstring could still add extra information that you can't capture in a type hint: describe what the parameter is used for and why a particular return value is chosen. You are right that I shouldn't just be restating the type hint in a docstring, that doesn't add any value. I'll keep this in mind in the future as well, I think I had in my mind that we always need a full docstring, but I agree that type hints may often make that not necessary. :+1: